checkers/mongo_test.go

```diff
@@ -0,0 +1,125 @@
+package checkers
+
+import (
+	"fmt"
+	. "github.com/onsi/gomega"
+	"github.com/zaffka/mongodb-boltdb-mock/db"
+	"testing"
+)
+
+func TestNewMongo(t *testing.T) {
+	RegisterTestingT(t)
+
+	t.Run("Happy path", func(t *testing.T) {
+		mongo := db.New(&db.Mock{})
+		url := "localhost:27017"
+		err := mongo.Connect(url)
+		defer mongo.Close()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	t.Run("Bad config should error", func(t *testing.T) {
+		var cfg *MongoConfig
+		r, err := NewMongo(cfg)
+
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("unable to validate mongodb config"))
+		Expect(r).To(BeNil())
+	})
+
+	t.Run("Should error when mongo server is not available", func(t *testing.T) {
+		cfg := &MongoConfig{
+			Ping: true,
+			Auth: &MongoAuthConfig{
+				Url: "foobar:42848",
+			},
+		}
+
+		r, err := NewMongo(cfg)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("no reachable servers"))
+		Expect(r).To(BeNil())
+	})
+}
+
+func TestValidateMongoConfig(t *testing.T) {
```

Comment:
Please add test case for invalid auth url as per: https://codecov.io/gh/InVisionApp/go-health/pull/51/diff?src=pr&el=tree#D3-93

Reply:
done